compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -2163,4 +2167,327 @@ LogicalResult OnlineAttentionOp::getResultTilePosition(
   return success();
 }
 
+//===---------------------------------------------------------------------===//
+// CustomOp
+//===---------------------------------------------------------------------===//
+
+/// These methods copied/modified from `TilingInterface` implementation of
+/// `getIterationDomain` of `LinalgOp`s.
+
+static SmallVector<OpFoldResult>
+createFlatListOfOperandDims(OpBuilder &builder, Location loc,
+                            CustomOp customOp) {
+  SmallVector<OpFoldResult> result;
+  for (Value operand : customOp->getOperands()) {
+    for (auto dim : llvm::seq<unsigned>(customOp.getRank(operand))) {
+      result.push_back(linalg::createFoldedDimOp(builder, loc, operand, dim));
```

Comment:
We have similar functions in LinalgExt. You can use `getDims`, e.g.,

```cpp
auto dims = getDims(builder, loc, operand);
result.append(dims.begin(), dims.end());
```

https://github.com/iree-org/iree/blob/691aa4ff5b9d4b96ccafde84016e7e3b96d6b9eb/compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.h#L27-L30